schema/documents.go

```diff
@@ -5,4 +5,5 @@ type Document struct {
 	PageContent string
 	Metadata    map[string]any
 	Score       float32
+	CustomID    string
```

Comment:
Starting here, it's not clear why `CustomID` is needed if we already have the `Metadata` field. Can't a custom ID be placed in metadata, if needed?

Reply:
Good point. As there is already a field `custom_id` in the database, I opted for making that field useful.

The other option is that stuff like this goes into the metadata (which is fine as well to me), but then I suggest we remove the `custom_id` column from the database. It doesn't make sense from my point of view to have a column in the database that is filled with random stuff that is unused (it just add clutter). Removing that column is probably even the best way forward as it avoids that similar stuff can be done in different ways.

The only risk we take there is that people might be relying on this column already which would cause existing implementations to break.

If we leave the column in, we should ensure that you can actually use it to do something meaningful.